### PR TITLE
Support passing the cmdline via "-e" to wezterm

### PR DESCRIPTION
**Pull request by vimpostor**
The easiest way to get this working with `clap` was to implement this using a dummy `-e` option.

This might look hacky at first, but it seems to be working quite reliably, for example the following command will still do the expected thing:

```bash
wezterm start -e echo --position 10,10
```

That is, it will echo "--position 10,10" and **not** parse it as a wezterm argument.

In the second commit I remove the "--" from the desktop file, which should be a better default, given how it improves the compatibility.

### CODE REVIEW

File: wezterm-gui-subcommands/src/lib.rs [hunk 1]
```diff
@@ -163,6 +163,18 @@ pub struct StartCommand {
...
+    /// Dummy argument that consumes "-e" and does nothing.
+    /// This is meant as a compatibility layer for supporting the
+    /// widely adopted standard of passing the command to execute
+    /// to the terminal via a "-e" option.
+    /// This works because we then treat the remaining cmdline as
+    /// trailing options, that will automatically be parsed via the
+    /// existing "prog" option.
+    /// This option exists only as a fallback. It is recommended to pass
+    /// the command as a normal trailing command instead if possible.
+    #[arg(long = "command", short = 'e', hide = true)]
```
Comment (wez) [inline wezterm-gui-subcommands/src/lib.rs:175]: Do we also need `--command`? I'd prefer not to use it for this if we don't need it.

```suggestion
    #[arg(short = 'e', hide = true)]
```

File: wezterm-gui-subcommands/src/lib.rs [hunk 2]
```diff
@@ -163,6 +163,18 @@ pub struct StartCommand {
...
+    pub cmd: bool,
```
Comment (wez) [inline wezterm-gui-subcommands/src/lib.rs:176]:
```suggestion
    _cmd: bool,
```

File: wezterm-gui/src/main.rs [hunk 3]
```diff
@@ -139,6 +139,7 @@ async fn async_run_ssh(opts: SshCommand) -> anyhow::Result<()> {
         always_new_process: true,
         class: opts.class,
         cwd: None,
+        cmd: false,
```
Comment (wez) [inline wezterm-gui/src/main.rs:142]: maybe replace with `.. Default::default()` construction, so we can avoid referencing the dummy arg?

File: wezterm-gui-subcommands/src/lib.rs (hunk 1 above)
Comment (vimpostor) [reply wezterm-gui-subcommands/src/lib.rs]: I agree with you.

File: wezterm-gui-subcommands/src/lib.rs (hunk 2 above)
Comment (vimpostor) [reply wezterm-gui-subcommands/src/lib.rs]: Removing the `pub` won't be possible, as neither the `Default::default()` nor the explicit initialization in wezterm-gui/src/main.rs will work then.

**Comment by wez** [comment]
Thanks!

**Pull request merged**